thunder/tests/test_transforms.py

```diff
@@ -437,6 +440,46 @@ def forward(self, x):
     assert t7 not in bwd_bsym_out, "Unexpected tensor rematerialized in the backward."
 
 
+def test_not_rematerialize_matmul():
+    nn = torch.nn
+
+    class MLP(nn.Module):
+        def __init__(self, embedding_size: int) -> None:
+            super().__init__()
+            self.layers = nn.Sequential(
+                nn.Linear(embedding_size, embedding_size * 4),
+                nn.GELU(),
+                nn.Linear(embedding_size * 4, embedding_size),
+                nn.Dropout(),
+            )
+
+        def forward(self, x):
+            return self.layers(x)
+
+    batch_size, sequence_length, embedding_size = 2, 3, 4
+    inp = torch.randn(batch_size, sequence_length, embedding_size, device="cuda", requires_grad=True)
+
+    model = MLP(embedding_size)
+    model.to("cuda")
+
+    jmodel = thunder.jit(model, nv_enable_linear=True, nv_enable_matmul=True)
```

Comment:
I'm not sure why nv_enable_linear and nv_enable_matmul need to be on to repro. I believe it's related to [this is_fusion check](https://github.com/Lightning-AI/lightning-thunder/blob/40904c762f6b1c2eb1a0c3a75a7b4b1382108b84/thunder/core/rematerialization.py#L239). @IvanYashchuk, feel free to code-comment an explanation.

Reply:
Without those are enabled matmul and linear are not fused into nvfuser regions and we have set them off by default since they were added